apps/dashboard/src/components/engine/status/components/Healthcheck.tsx

```diff
@@ -0,0 +1,117 @@
+"use client";
+
+import { Skeleton } from "@/components/ui/skeleton";
+import { ToolTipLabel } from "@/components/ui/tooltip";
+import { useQuery } from "@tanstack/react-query";
+import { PrimaryInfoItem } from "app/(dashboard)/(chain)/[chain_id]/(chainPage)/components/server/primary-info-item";
+import { CircleCheck, XIcon } from "lucide-react";
+import { useState } from "react";
+
+function useChainStatswithRPC(engineUrlBase: string) {
+  const [shouldRefetch, setShouldRefetch] = useState(true);
+  const engineUrl = `${engineUrlBase}/system/health`;
+  return useQuery({
+    queryKey: ["engine-stats", { engineUrl }],
+    queryFn: async () => {
+      const startTimeStamp = performance.now();
+      const res = await fetch(engineUrl);
+
+      const json = await res.json();
+      const latency = (performance.now() - startTimeStamp).toFixed(0);
+
+      return {
+        latency,
+        status: json.status,
+      };
+    },
+    refetchInterval: shouldRefetch ? 5 * 1000 : undefined,
+    enabled: !!engineUrl,
+    refetchOnWindowFocus: false,
+    onError: () => {
+      setShouldRefetch(false);
+    },
```

Comment:
why would you want to stop checking if it's down? feels like we should keep checking to see if it comes back up?